frontends/web/src/routes/buy/exchange.tsx

```diff
@@ -14,44 +14,203 @@
  * limitations under the License.
  */
 
-import { Button } from '../../components/forms';
-import { isPocketSupported, isMoonpayBuySupported } from '../../api/exchanges';
+import { useTranslation } from 'react-i18next';
+import React, { useState, useEffect } from 'react';
+import { Button, Select, Radio } from '../../components/forms';
+import * as exchangesAPI from '../../api/exchanges';
+import { IAccount } from '../../api/account';
+import { Header } from '../../components/layout';
+import Guide from './guide';
+import { findAccount, getCryptoName } from '../account/utils';
 import { route } from '../../utils/route';
 import { useLoad } from '../../hooks/api';
+import { languageFromConfig, localeMainLanguage } from '../../i18n/config';
+import style from './exchange.module.css';
 
-interface TProps {
+type TProps = {
     code: string;
+    accounts: IAccount[];
+
+}
+
+type TOption = {
+    text: string;
+    value?: string;
 }
 
 // TODO:
 // - add layout
-export const Exchange = ({ code }: TProps) => {
-  const showPocket = useLoad(isPocketSupported(code));
-  const showMoonpay = useLoad(isMoonpayBuySupported(code));
+export const Exchange = ({ code, accounts }: TProps) => {
+  const { t } = useTranslation();
+
+  const [showPocket, setShowPocket] = useState(false);
+  const [showMoonpay, setShowMoonpay] = useState(false);
+  const [selectedRegion, setSelectedRegion] = useState('');
+  const [selectedExchange, setSelectedExchange] = useState('');
+  const [regions, setRegions] = useState<TOption[]>([]);
+  const [locale, setLocale] = useState('');
+  const [bestDeal, setBestDeal] = useState<exchangesAPI.ExchangeDeal>();
+
+  const isPocketSupportedCoin = useLoad(exchangesAPI.isPocketSupported(code));
+  const isMoonpaySupportedCoin = useLoad(exchangesAPI.isMoonpayBuySupported(code));
```

Comment:
one more thing, iirc useLoad by default only fires once, so if code changes for some reason it will not requst again.

Found this again when looking at the other PR that tests the hooks.

If this is a problem, i.e. if code can change at anytime (but I think it doesn't right?) then you would need to specify a 2nd arg and pass `[code]` as dependencies so it can watch for changes in the dependencyList and make a new request if code changed.

Reply:
I don't think that the code can change inside this page, should be ok as is